src/main/scala/tech/cryptonomic/conseil/util/ConfigUtil.scala

```diff
@@ -119,45 +125,57 @@ object ConfigUtil {
-        * @param namespace the path where the custom configuration will be searched-for
-        */
-      def loadAkkaStreamingClientConfig(namespace: String): Either[ConfigReaderFailures, HttpStreamingConfiguration] = {
-
-        //this is where akka searches for the config entry for host connection pool
-        val referenceHostPoolEntryPath = "akka.http.host-connection-pool"
-
-        //read a conseil-specific entry into the expected path for the http streaming client host pool
-        def loadHostPoolConfig(rootConfig: Config): Either[ConfigReaderFailures, Config] =
-          Try(rootConfig.getConfig(namespace)
-                .atPath(referenceHostPoolEntryPath) //puts the config entry where expected by akka
-                .withFallback(rootConfig) //adds default values, where not overriden
-                .ensuring(
-                  endConfig =>
-                    //verifies all expected entries are there
-                    Try(endConfig.checkValid(rootConfig.getConfig(referenceHostPoolEntryPath), referenceHostPoolEntryPath)).isSuccess
-                )
-            ).toEither
-            .left.map{
-              //wraps the error into pureconfig's one
-              t => ConfigReaderFailures(new ThrowableFailure(t, None))
-            }
+    /**
+      * Reads a specific entry in the configuration file, to create a valid akka-http client host-pool configuration
+      *
+      * @param namespace the path where the custom configuration will be searched-for
+      */
+    def loadAkkaStreamingClientConfig(namespace: String): Either[ConfigReaderFailures, HttpStreamingConfiguration] = {
+      // this is where akka searches for the config entry for host connection pool
+      loadConfigForEntryPath(namespace, "akka.http.host-connection-pool").map(HttpStreamingConfiguration)
+    }
 
-        //compose the configuration reading outcomes
-        for {
-          akkaConf <- pureconfig.loadConfig[Config]
-          tezosClientConfig <- loadHostPoolConfig(akkaConf)
-        } yield HttpStreamingConfiguration(tezosClientConfig)
-      }
+    /**
+      * Reads a specific entry in the configuration file, to create a valid akka-http caching configuration
+      *
+      * @param namespace the path where the custom configuration will be searched-for
+      */
+    def loadAkkaCacheConfig(namespace: String): Either[ConfigReaderFailures, HttpCacheConfiguration] = {
+      // this is where akka searches for the config entry for host connection pool
+      loadConfigForEntryPath(namespace, "akka.http.caching.lfu-cache").map(HttpCacheConfiguration)
+    }
+
+    private def loadConfigForEntryPath(namespace: String, referenceEntryPath: String): Either[ConfigReaderFailures, Config] = {
+      //read a conseil-specific entry into the expected path for the config
+      def loadHostPoolConfig(rootConfig: Config): Either[ConfigReaderFailures, Config] =
```

Comment:
I think this might need renaming, since now it's path-agnostic and not specific to host-pool...

Reply:
renamed, please tell if it's ok